eng/pipelines/templates/jobs/ci.yml

```diff
@@ -190,6 +190,40 @@ jobs:
           failOnStderr: false
       - template: /eng/common/pipelines/templates/steps/eng-common-workflow-enforcer.yml
 
+  - job: "ComplianceTools"
+    pool:
+      name: azsdk-pool-mms-win-2019-general
+      vmImage: MMS2019
+    steps:      
+      - pwsh: |
+          New-Item -Path "$(Build.sourcesdirectory)" -Name "changedFiles" -ItemType "directory"
+          Write-Host "$(System.PullRequest.TargetBranch)"
+          git diff "$(System.PullRequest.TargetBranch)" HEAD --name-only | ForEach-Object {Copy-Item -Path $_ -Destination "$(Build.sourcesdirectory)/changedFiles"}
+          Get-ChildItem "$(Build.sourcesdirectory)/changedFiles"
+        displayName: Put changed files to folder
+        condition: eq(variables['Build.Reason'],'PullRequest')
+      - task: securedevelopmentteam.vss-secure-development-tools.build-task-credscan.CredScan@2
+        displayName: Scanning for credentials
+        inputs:
+          toolMajorVersion: V2
+          toolVersion: latest
+          ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
```

Comment:
I wouldn't do this conditioning here I would keep it all in the pre-step above and have this only read a variable that contains the list of directories to scan.